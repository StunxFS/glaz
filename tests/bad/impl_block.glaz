struct St { }

impl St {
	const INVALID: u8 = 0;
}

enum En { X }

impl En {
	// invalid
	fn assoc_fn() {}
}

fn main() {}
