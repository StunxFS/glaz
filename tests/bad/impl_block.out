tests/bad/impl_block.glaz:3:7: error: `impl` blocks can only contain functions and methods
     | 
   3 | impl St {
     |      ^^
tests/bad/impl_block.glaz:11:6: error: cannot add associated functions to an enum
     | 
  11 | 	fn assoc_fn() {}
     | 	   ^^^^^^^^^^
glaz error: could not compile `impl_block`, aborting due to 2 previous errors.
